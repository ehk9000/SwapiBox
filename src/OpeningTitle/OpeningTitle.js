import React, {Component} from 'react';
import './OpeningTitle.scss'
import rebel from '../images/rebel-alliance.svg'



class OpeningTitle extends Component {
    constructor() {
        super();
        this.state = {
            randomFilm: [],
            isLoading: false
        }
    }

    async componentDidMount() {
        const index = Math.floor(Math.random() * 7) + 1;
        this.setState({isLoading: true})
        const url = `https://swapi.co/api/films/${index}`;
        const response = await fetch(url);
        const randomFilm = await response.json();
        this.setState({randomFilm, isLoading: false})
        console.log(this.state.randomFilm)
    }

    render() {
        const {opening_crawl: openingCrawl, title, release_date: date} = this.state.randomFilm;
        let openingContainer = this.state.isLoading ? 
        <div className="loading-wrapper">
            <img src={rebel} alt="Rebel Alliance Icon" className="rebel" />
            <p className="load" >Loading...</p>
            <img src={rebel} alt="Rebel Alliance Icon" className="rebel" />
        </div>
        :
        <article className="fade">
            <section className="star-wars">
                <div className="crawl">
                    <p>{openingCrawl}</p>
                    <h2>{title}</h2>
                    <p>{date}</p>
                </div>
            </section>  
        </article>

        return (
            <div>
              {openingContainer}
            </div>
        )
    }
}




export default OpeningTitle