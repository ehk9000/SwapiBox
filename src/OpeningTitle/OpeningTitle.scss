.fade {
    position: relative;
    width: 100%;
    min-height: 60vh;
    top: -25px;
    z-index: 1;
  }

.crawl {
    position: relative;
    top: 9999px;
    transform-origin: 20% 100%;
    animation: crawl 60s linear;
}

.star-wars {
    display: flex;
    justify-content: center;
    position: relative;
    height: 800px;
    color: #feda4a;
    font-family: 'Pathway Gothic One', sans-serif;
    font-size: 500%;
    font-weight: 600;
    letter-spacing: 6px;
    line-height: 150%;
    perspective: 400px;
    text-align: justify;
  }

  .rebel {
      height: 115px;
      margin: 20px;
  }

  .loading-wrapper {
      display: flex;
      justify-content: center
  }

  .load {
      font-size: 50px;
      color:slategrey;
  }

  .remove {
      display: none;
  }

  .skip-btn {
      z-index: 4;
  }

@keyframes crawl {
    0% {
        top: 0;
        transform:rotateX(20deg) translateZ(0)
    }
    100% {
        top: -6000px;
        transform:rotateX(25deg) translateZ(-2500px)
    }
}