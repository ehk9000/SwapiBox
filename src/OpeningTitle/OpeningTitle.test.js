import React from 'react';
import {shallow} from 'enzyme';
import OpeningTitle from './OpeningTitle';
import {stringify} from 'querystring';

// mock random film 
// mock fetch


describe('OpeningTitle', () => {
    let wrapper;
    let mockFilm;

    beforeEach(() => {
        mockFilm = {title: "Revenge of the Sith", 
        opening_crawl:"War! The Republic is crumbling under attacks by the ruthlessSith Lord...",
        release_date: "2005-05-19"};
        wrapper = shallow(<OpeningTitle />)
        window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
            json: () => Promise.resolve(mockFilm)
        }))
    })
    it('should match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
})