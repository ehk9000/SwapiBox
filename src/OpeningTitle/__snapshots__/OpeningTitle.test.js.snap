// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpeningTitle should match Snapshot 1`] = `ShallowWrapper {}`;
